### ISSUE TITLE
Clarify flex/main v. master(default) branch -- how to "compile" a recipe?

### ISSUE THREAD

**Issue by tacman** [symfony/recipes#1072]
I've developed a bundle and am trying to create a recipe.  First I want to test it locally, so I follow the instructions at https://symfony.com/doc/current/setup/flex_private_recipes.html -- not because I want a private recipe (that is, neither the bundle nor the recipe is "private" in the sense of a github project.)

I have read, re-read, then read again the documentation for creating a recipe, and I feel like I'm missing a very important step.  The documentation for a private repository only talks about the package.version.json file, e.g. 

```json
{
    "manifests": {
        "acme/private-bundle": {
            "manifest": {
                "bundles": {
                    "Acme\\PrivateBundle\\AcmePrivateBundle": [
                        "all"
                    ]
                },
                "copy-from-recipe": {
                    "config/": "%CONFIG_DIR%"
                }
            },
            "files": {
                "config/packages/acme_private.yaml": {
                    "contents": [
                        "acme_private:",
                        "    encode: true",
                        ""
                    ],
                    "executable": false
                }
            },
            "ref": "7405f3af1312d1f9121afed4dddef636c6c7ff00"
        }
    }
}
```

This file appears to be the output of some command that reads the file in the contrib format, that is, a directory structure that has a config directory for routes and packages, maybe a src directory, etc.  Obviously no one is typing in by hand line by line a yaml file (or PHP!) as a list of strings in the "content" key.

It appears that the master branch contains the directory structure, and flex/main branch contains the "compiled" json recipe file.  But in the private recipe instructions, there's no instructions as to how to get from the DX-friendly files to the JSON recipe.

TL;DR; How do I create a package.version.json recipe from the package/version/ directory that contains manifest.json, config, etc.?

**Comment by DjLeChuck** [comment]
Hi,

I was asking myself the same question today and after a bit of investigating, I think it's done with an automated task: https://github.com/symfony/recipes/blob/master/.github/workflows/flex-update.yml

This task relies on [symfony-tools/recipes-checker](https://github.com/symfony-tools/recipes-checker/)

**Comment by nicolas-grekas** [comment]
To test a recipe, submit it here, and a bot will give your easy instructions to do so.
Running private recipes is something that needs more doc for now.
